test/e2e/providers/vsphere/vsphere.go

```diff
@@ -52,19 +66,44 @@ func newVSphereMachineProviderSpec(clusterID string) (*vsphere.VSphereMachinePro
 		},
 		NumCPUs:           int32(4),
 		NumCoresPerSocket: int32(1),
-		// The template is hardcoded with an image which has been properly sysprepped.
-		// TODO: Find a way to automatically update this with latest image
-		Template: "windows-golden-images/windows-server-2004-template",
-		Workspace: &vsphere.Workspace{
-			Datacenter:   "SDDC-Datacenter",
-			Datastore:    "WorkloadDatastore",
-			Folder:       "/SDDC-Datacenter/vm/" + clusterID,
-			ResourcePool: "/SDDC-Datacenter/host/Cluster-1/Resources",
-			Server:       "vcenter.sddc-44-236-21-251.vmwarevmc.com",
-		},
+
+		Template:  vmTemplate,
+		Workspace: workspace,
 	}, nil
 }
 
+// getWorkspaceFromExistingMachineSet returns Workspace from a machineset provisioned during installation
+func (p *Provider) getWorkspaceFromExistingMachineSet() (*vsphere.Workspace, error) {
+	clusterID, err := p.oc.GetInfrastructureID()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get cluster id: %v", err)
```

Comment:
We've been following the pattern of using errors.Wrap like so:
```
	clusterID, err := p.oc.GetInfrastructureID()
	if err != nil {
		return nil, errors.Wrap(err, "unable to get cluster id")
```

To stay consistent could you also follow this pattern? Applies to other lines as well